File: functions/python/get_oui.py
def get_oui(mac, data):
    formatted_mac = mac.replace("-", "").replace(":", "").upper()[:6]
    oui_info = data["ouis"].get(formatted_mac)
    
    if not oui_info:
        return None
    
    return {
        "mac": formatted_mac,
        "country": data["countries"].get(oui_info["cc"]),
        "organization": data["organizations"].get(oui_info["org"])
    }

File: functions/javascript/getOUI.js
function getOUI(mac, data) {
    const formattedMac = mac.replace(/[-:]/g, '').substring(0, 6).toUpperCase();
    const ouiInfo = data.ouis[formattedMac];
  
    if (!ouiInfo) return null;
  
    return {
      "mac": formattedMac,
      "country": data.countries[ouiInfo.cc],
      "organization": data.organizations[ouiInfo.org]
    };
  }